### ISSUE TITLE
Upgrading to 6.1.0 makes react-fontawesome incompatible

### ISSUE THREAD

**Issue by lukesrw** [FortAwesome/Font-Awesome#18823]
Per documentation, https://fontawesome.com/v6/docs/web/use-with/react/add-icons#add-individual-icons-explicitly

Previously working, now broken, example:

```tsx
import ReactDOM from 'react-dom'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faCoffee } from '@fortawesome/pro-duotone-svg-icons'

const element = <FontAwesomeIcon icon={faCoffee} />

ReactDOM.render(element, document.body)
```

The "icon" prop of FontAwesomeIcon requires type `IconProp`.

Previously icons exported from `pro-duotone-svg-icons` were of type `IconProp`.

However in the latest version they are of type `IconDefinition`.

**Comment by robmadole** [comment]
@lukesrw did you upgrade `@fortawesome/fontawesome-svg-core` and `@fortawesome/fontawesome-common-types` to `6.1.0` as well?

**Comment by lukesrw** [comment]
`@fortawesome/fontawesome-common-types` isn't a dependency.

`@fortawesome/fontawesome-svg-core` was listed at version 1.3.0, I didn't expect it to jump 5 major versions.

I've updated to 6.1.0 and it seems to work, shouldn't `react-fontawesome` be updated to list 6.1.0 as a minimum dependency?

**Comment by redkevvy** [comment]
Similarly the angular package is having issues, similar if not identical here and it broke at least one build (likely our fault for being too edgy).

We also noticed that ```@fortawesome/fontawesome-svg-core``` did jump from version 1.3.0 to 6.1.0.
The other packages were at:
```@fortawesome/free-solid-svg-icons: "^6.0.0"```
```@fortawesome/pro-regular-svg-icons: "^6.0.0"```

Only difference is we're running with ```@fortawesome/angular-fontawesome``` @ 0.10.1, and not react.

Update: To quell the immediate issue I just explicitly removed the semver markings to reference the older specific packages, i.e. removed carat (^) and tilde (~) for all fortawesome modules.  It's a stop-gap in the meantime.

**Comment by robmadole** [comment]
We've aligned all the versions for `@fortawesome/fontawesome-common-types`, `@fortawesome/fontawesome-svg-core`, and `@fortawesome/*-svg-icons`. We were being too clever before and it was just plain confusing. So that jump in the version was just to align it with the other packages.

I've released `0.0.18` of the react-fontawesome component which updates the peer dependencies to include `~6`. That should help.

@redkevvy I believe the maintainer of angular-fontawesome is planning on addressing this soon.
